pkg/internal/controller/controller.go

```
@@ -468,9 +471,6 @@ func (c *Controller[request]) reconcileHandler(ctx context.Context, req request,
 		}
 		ctrlmetrics.ReconcileErrors.WithLabelValues(c.Name).Inc()
 		ctrlmetrics.ReconcileTotal.WithLabelValues(c.Name, labelError).Inc()
-		if !result.IsZero() {
```

If Priority is set result is not zero anymore. Not ideal, but I don't have a better idea.

Independent of that I think it's not especially bad to get rid of this warning.
I've written code specifically to set ctrl.Result{} to zero to avoid this warning which would have been absolutely not necessary otherwise

---

> Could explicitly check for Requeue or RequeAfter being set.

I'll do that

> It is a bit awkward that we now respect a non-error return when an error is returned, that is very uncommon

I think we don't "respect" it (in the sense of using it), we just don't write a warning on log level 0.

We have a bunch of controllers where the main reconcile logic returns either an error or non-zero ctrl.Result{}, but then we patch the reconciled object with defer. If that patch fails we want to return the error.

Because of this code here in controller-runtime we have to make sure that in these situations we set Result to zero to avoid this warning: https://github.com/kubernetes-sigs/cluster-api/blob/main/bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go#L274-L276

Not sure I see the value of this warning over CR just silently prioritizing error over ctrl.Result{} if both is returned